### ISSUE TITLE
Menu needs to support static and dynamic items

### ISSUE THREAD

**Issue by j0nscalet** [j0nscalet/QualityKitchen.Mobile#9]
<img src="https://cloud.githubusercontent.com/assets/1328795/24057408/83973724-0b04-11e7-8dad-4d5ce03f0f8b.JPG" width="75%" height="75%">

 **Problem**
The menu needs to support different item types. It needs to support:
- MenuItem - an item that contains a text description and an icon, on tap it should trigger an action
- BubbleItem - an item that contains a text description and bubble which contains a count, on tap it should trigger an action
- ContentItem - *In the future* an item that contains a title and a body; it does not respond to a tap 

https://mallibone.com/post/xamarin.forms-grouped-listview
https://www.cocoanetics.com/2011/03/expandingcollapsing-tableview-sections/
https://github.com/OliverLetterer/SLExpandableTableView

**Questions**
- How is the Menu ItemsSource set?
   - This should be built using a ListView

- How do we bind the menu from outside properties like Checklists or Incidents?
   - Do we use BindingExpressions? 
   - Is there an object source and property for each menu item?
   - [Still questions that need to be answered on this issue]
```
App.Menu.ItemSource = new List<MenuItem>(){
   new BubbleItem(){
      Text="Grill",
      BubbleCount=20
   },
  new BubbleItem(){
     Text="Prep",
     BubbleCount=4
  },
  new BubbleItem(){
     Text="Expo",
     BubbleCount=12
  },
  new MenuItem(){
     Icon="Warning"
     Text="Incident Center",
      Action=() =>{
      }
  }
}
```

- Will OnSelected(object item) method cast correctly?
```
public override DataTemplate SelectTemplate (object item, BindableObject container){
    var menuItem = item as MenuItem;
    // Can I do something like this?
    if(menuItem is MenuItem)
   
   // If not we'll need to add a type property to MenuItem
}
```

**Solution**
1. Define classes that model the MenuItems
```
public class MenuItem{
   string Description
   Image Icon;
   Action Tapped;
}

public class BubbleItem : MenuItem{
   int BubbleCount
}
```
2. Create Menu.xaml view
3. Create 'MenuViewModel'
4. Create 'MenuViewModel.Refresh()' to refresh the menu

**Comment by j0nscalet** [comment]
For reference when we circle back to this.
![image](https://cloud.githubusercontent.com/assets/1328795/24383139/10d01d00-1310-11e7-8328-ddef55e7357c.png)

**Comment by j0nscalet** [comment]
Note to self: We're going to need to turn off 'Row Animations' in the ListView control. [PR#733](https://github.com/xamarin/Xamarin.Forms/pull/733) in XForms accomplishes and [has been merged into beta-2.3.5-pre1](https://github.com/xamarin/Xamarin.Forms/commit/c84bb83676d3f1898d6419447e103d27e1435374)
